reactive-streams/src/main/scala/fs2/interop/reactivestreams/StreamSubscriber.scala

```diff
@@ -72,23 +74,23 @@ final class StreamSubscriber[F[_]: ConcurrentEffect, A](val sub: StreamSubscribe
 }
 
 object StreamSubscriber {
-  def apply[F[_]: ConcurrentEffect, A]: F[StreamSubscriber[F, A]] =
+  def apply[F[_]: Async, A]: F[StreamSubscriber[F, A]] =
     fsm[F, A].map(new StreamSubscriber(_))
 
   /** A finite state machine describing the subscriber */
   private[reactivestreams] trait FSM[F[_], A] {
 
     /** receives a subscription from upstream */
-    def onSubscribe(s: Subscription): F[Unit]
+    def onSubscribe(s: Subscription): Unit
 
     /** receives next record from upstream */
-    def onNext(a: A): F[Unit]
+    def onNext(a: A): Unit
 
     /** receives error from upstream */
-    def onError(t: Throwable): F[Unit]
+    def onError(t: Throwable): Unit
 
     /** called when upstream has finished sending records */
-    def onComplete: F[Unit]
+    def onComplete(): Unit
```

Comment:
These are also binary incompatible, but you can probably get away with exemptions here since it's package-private.

Reply:
Yeah, I think we're fine here.